include/CXXGraph/Graph/Graph_impl.hpp

```diff
@@ -232,22 +232,23 @@ void Graph<T>::removeEdge(const CXXGraph::id_t edgeId) {
 template <typename T>
 void Graph<T>::removeNode(const std::string &nodeUserId) {
   auto nodeOpt = getNode(nodeUserId);
-  auto isolatedNodeIt = isolatedNodesSet.find(nodeOpt.value());
-
-  if (nodeOpt.has_value() && isolatedNodeIt != isolatedNodesSet.end()) {
-    // The node is isolated
-    isolatedNodesSet.erase(isolatedNodeIt);
-  } else if (nodeOpt.has_value()) {
-    // The node is not isolated
-    // Remove the edges containing the node
-    auto edgeset = edgeSet;
-    for (auto edgeIt : edgeset) {
-      if (edgeIt->getNodePair().first->getUserId() == nodeUserId ||
-          edgeIt->getNodePair().second->getUserId() == nodeUserId) {
-        this->removeEdge(edgeIt->getId());
+  if (nodeOpt) {
+    auto isolatedNodeIt  = isolatedNodesSet.find(nodeOpt.value());
+    if (isolatedNodeIt != isolatedNodesSet.end()) {
+      // The node is isolated
+      isolatedNodesSet.erase(isolatedNodeIt);
+    } else {
+      // The node is not isolated
+      // Remove the edges containing the node
+      auto edgeset = edgeSet;
+      for (auto edgeIt : edgeset) {
+        if (edgeIt->getNodePair().first->getUserId() == nodeUserId ||
+            edgeIt->getNodePair().second->getUserId() == nodeUserId) {
+          this->removeEdge(edgeIt->getId());
```

Comment:
Here you loop over a copy of the `edgeSet`, but don't modify it's content, since you erase the node directly from the class set through `this`, so you should declare `edgeset` as `const auto& edgeset = edgeSet`